plugins/analytics-module-snowplow/src/apis/implementations/AnalyticsApi/SnowplowAnalytics.ts

```diff
@@ -1,127 +1,164 @@
+    enabled: boolean;
+    baseUrl: string;
+    trackerId: string;
+    endpoint: string;
+    appId: string;
+    cookieLifetime: number;
+    debounceTime: number;
+  }) {
+    const {
+      enabled,
+      baseUrl,
+      trackerId,
+      endpoint,
+      appId,
+      cookieLifetime,
+      debounceTime,
+    } = options;
 
-        this.enabled = enabled;
-        this.baseUrl = baseUrl;
-        this.cancelProc = null;
-        this.debounceTime = debounceTime;
-        
-        // create the Snowplow tracker
-        if (this.enabled) {
-            newTracker(trackerId, endpoint, {
-                appId: appId,
-                cookieLifetime: cookieLifetime,
-                platform: "web",
-                contexts: {
-                    webPage: true
-                },
-                plugins: [LinkClickTrackingPlugin(), SiteTrackingPlugin()]
-            });
+    this.enabled = enabled;
+    this.baseUrl = baseUrl;
+    this.cancelProc = null;
+    this.debounceTime = debounceTime;
 
-            enableActivityTracking({
-                minimumVisitLength: 30,
-                heartbeatDelay: 30
-            });
-        }
-    }
-  
-    static fromConfig(config: Config): SnowplowAnalytics {
-        const enabled = config.getBoolean('app.analytics.snowplow.enabled');
-        const baseUrl = config.getString('app.baseUrl');
-        const endpoint = config.getString('app.analytics.snowplow.collectorUrl');
-        const appId = config.getOptionalString('app.analytics.snowplow.appId') || 'Snowplow_standalone_OCIO' ;
-        const trackerId = config.getOptionalString('app.analytics.snowplow.trackerId') || 'rt';
-        const cookieLifetime = config.getOptionalNumber('app.analytics.snowplow.cookieLifetime') ?? 86400 * 548;
-        const debounceTime = config.getOptionalNumber('app.analytics.snowplow.debounceTime') ?? 3000;
+    // create the Snowplow tracker
+    if (this.enabled) {
+      newTracker(trackerId, endpoint, {
+        appId: appId,
+        cookieLifetime: cookieLifetime,
+        platform: 'web',
+        contexts: {
+          webPage: true,
+        },
+        plugins: [linkClickTrackingPlugin(), siteTrackingPlugin()],
+      });
 
-        return new SnowplowAnalytics({
-            enabled,
-            baseUrl,
-            trackerId,
-            endpoint,
-            appId,
-            cookieLifetime,
-            debounceTime
-        });
-    }
-    
-    captureEvent(event: AnalyticsEvent): void {
-        if (this.enabled) {
-            switch (event.action) {
-                case "search":
-                    this.captureSearch(event);
-                    break;
-                case "navigate":
-                    trackPageView();
-                    break;
-                case "click":
-                case "discover":
-                    this.trackClick(event);
-                    break;
-            }
-        }
+      enableActivityTracking({
+        minimumVisitLength: 30,
```

Comment:
curious what this is doing!

Reply:
The activity tracking cause page pings to fire when users are engaged on the page.
So, if a user has the page focused and they're scrolling or moving their mouse then the ping will go off every 30 seconds. If there's no activity then the pings stop.
I'm not sure if this metric is visible in the dashboards though.